lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py

```diff
@@ -99,6 +99,14 @@ def __init__(self, module):
         self.options = self.params.get('options', dict())
         self.hosts = self.get_all_host_objs(cluster_name=cluster_name, esxi_host_name=esxi_host_name)
 
+    @staticmethod
+    def is_integer(value, type_of='int'):
```

Comment:
This function is a good idea as it catches a potential error condition which the original PR did not consider, passing a value that is too large for an integer field.  I did not see any configuration options in vmware which allowed values outside of the range of an int, but this will prevent that issue from occurring if they ever do.

I have some small issue with this new function: Allowing the type to be passed in as a string is a little unexpected on an is_integer function.  This allows the function to be used for purposes that its name does not imply.  There is also lots of potential for typos as this module is changed in the future.   I don't have a good suggestion to "resolve" this except to make two functions is_long and is_int.

Reply:
I am thinking from Python 3 perspective, as `Int` and `Long` are merged into `Int` So function name `is_integer`. I wanted to avoid creating two functions `is_long` and `is_int`, but I am OK if it removes the confusion. 👍

Reply:
Since the function is for distinguishing between vmware types not python types and vmware will have int and long types for the foreseeable future, we should probably limit it to those two types or change it to something like is_vmware_type() or matches_vmware_type() if we want to keep the function the way it is.